### DISCUSSION TITLE
INVALID_IMAGE_FORMAT_DESCRIPTOR

### DISCUSSION THREAD

**Discussion by gybo03** [pyopencl/pyopencl#806] [Q&A] [answered]
case 1)
image = np.array(Image.open(image_path).convert("RGBA"))
image_buf = cl.image_from_array(context, image, 4)
case 2)
image = np.array(Image.open(image_path).convert("RGB"))
image_buf = cl.image_from_array(context, image, 3)

for context this is shape of an array:
image = np.array(Image.open(image_path))
image.shape=(h,w,3)

case 1 works fine without error but case 2 gives back the error:
pyopencl._cl.RuntimeError: clCreateImage failed: INVALID_IMAGE_FORMAT_DESCRIPTOR

in the documentation it is mentioned that rgb is supported i have seen it also in the code of image_from_array(). Im a nooby so i tried finding the Image class to see what i am missing but i couldnt find it. Why does it need that one more byte for alpha, what am i missing. I am using PIL image.

 - OS: [e.g. Linux] linux
 - ICD Loader and version: [e.g. ocl-icd 2.3.1] ocl-icd 2.3.2
 - ICD and version: [e.g. pocl 1.8] intel-compute-runtime
 - CPU/GPU: [e.g. Nvidia Titan V]intel uhd 620
  - Python version: [e.g. 3.10]3.12
  - PyOpenCL version: [e.g. 2021.1]2024.3

**Comment by inducer** [comment]
That error message comes from the ICD. Does it support RGB? Do the arguments that `image_from_array` passes to `create_image` seem plausible (maybe use a debugger to check)?

**Reply by gybo03** [reply, marked as answer]
i see now it seems that intel is not supporting any image type other than RGBA for some god forsaken reason. Maybe add that as note that intel igpus dont support fomrats other than:
Format [0] = CL_RGBA / CL_UNORM_INT8
Format [1] = CL_RGBA / CL_UNORM_INT16
Format [2] = CL_RGBA / CL_SIGNED_INT8
Format [3] = CL_RGBA / CL_SIGNED_INT16
Format [4] = CL_RGBA / CL_SIGNED_INT32
Format [5] = CL_RGBA / CL_UNSIGNED_INT8
Format [6] = CL_RGBA / CL_UNSIGNED_INT16
Format [7] = CL_RGBA / CL_UNSIGNED_INT32
Format [8] = CL_RGBA / CL_HALF_FLOAT
Format [9] = CL_RGBA / CL_FLOAT
Format [10] = CL_BGRA / CL_UNORM_INT8
Format [11] = CL_INTENSITY / CL_FLOAT
Format [12] = CL_LUMINANCE / CL_FLOAT
i found this [here](https://community.intel.com/t5/OpenCL-for-CPU/clCreateImage2D-always-returns-10-CL-IMAGE-FORMAT-NOT-SUPPORTED/m-p/775490)

**Reply by gybo03** [reply]
thanks for fast replies either way
